lib/filterx/tests/test_expr_plus_generator.c

```diff
@@ -233,35 +235,28 @@ Test(expr_plus_generator, test_nested_dict_add_variable_to_generator_with_post_s
                                                               filterx_literal_inner_dict_generator_new(lhs, lhs_inner), FALSE));
   filterx_literal_generator_set_elements(lhs, lhs_vals);
 
-  FilterXExpr *rhs = filterx_non_literal_new(filterx_json_object_new_from_repr("{\"tik\":{\"tak\":\"toe\"}}", -1));
+  FilterXExpr *rhs = filterx_non_literal_new(filterx_object_from_json("{\"tik\":{\"tak\":\"toe\"}}", -1, NULL));
 
   FilterXExpr *expr = filterx_operator_plus_generator_new(lhs, rhs);
-  FilterXObject *json_dict = filterx_json_object_new_empty();
-  FilterXExpr *fillable = filterx_literal_new(json_dict);
+  FilterXObject *dict = filterx_dict_new();
+  FilterXExpr *fillable = filterx_literal_new(dict);
```

Comment:
I think it would be worth adding the test case we talked about:

```
fx_var = dict($MSG);
b = fx_var.a;
b.c = 2;

fx_var, fx_var.a remains unchanged
```

```
b = $MSG.a;
$MSG.a.b.c = 2;

dirty tracking works ($MSG is synced), b remains unchanged 
```